pkg/setting/setting_test.go

```diff
@@ -296,6 +296,35 @@ func TestLoadingSettings(t *testing.T) {
 			So(value, ShouldEqual, "")
 		})
 	})
+
+	Convey("Test reading provisioning config", t, func() {
+		Convey("By default it should use provisioning folder setting", func() {
+			cfg := NewCfg()
+			err := cfg.Load(&CommandLineArgs{
+				HomePath: "../../",
+			})
+			So(err, ShouldBeNil)
+			So(cfg.ProvisioningDashboardsPath, ShouldEqual, "../../conf/provisioning/dashboards")
+			So(cfg.ProvisioningDatasourcesPath, ShouldEqual, "../../conf/provisioning/datasources")
+			So(cfg.ProvisioningNotifiersPath, ShouldEqual, "../../conf/provisioning/notifiers")
+			So(cfg.ProvisioningPluginsPath, ShouldEqual, "../../conf/provisioning/plugins")
+		})
+
+		Convey("With settings in provisioning section it should override defaults", func() {
+			cfg := NewCfg()
+			err := cfg.Load(&CommandLineArgs{
+				HomePath: "../../",
+				Config:   filepath.Join(HomePath, "pkg/setting/testdata/provisioning.ini"),
+			})
+			So(err, ShouldBeNil)
+			So(cfg.ProvisioningDashboardsPath, ShouldEqual, "/etc/absolute/path")
+			So(cfg.ProvisioningDatasourcesPath, ShouldEqual, "../../not/absolute/path")
+			// There are no entries in mock provisioning config for plugins and notifiers
+			// They should equal default values
```

Comment:
```suggestion
			// There are no entries in the test provisioning config for plugins and notifiers
			// They should equal the defaults
```

Reply:
Why aren't you configuring also datasources_path and dashboards_path in this test, so you can test also those settings? I think that would make more sense for a test case where you want to test non-defaults.

Reply:
I tried to cover two cases in the one test. 
1. All options are specified. 
2. Some options are specified. 
I think I will follow your suggestion because I was not 100% sure about this decision.

Reply:
@nabokihms Thanks for clarifying that, I figured it was your thinking. Personally, if I wanted to test the other case (of a mixture of defaults and non-defaults), I would make second test case. I don't think I want to ask you to do it though, since it should work and the PR already has good test coverage :)